server/controllers/auth_controller.js

```diff
@@ -33,18 +34,27 @@ const postRegister = (req, res, next) => {
 
   User.findOne({ username: props.username })
     .then(user => {
-      if (user) return next(createError({
-        status: CONFLICT,
-        message: 'Username already exists'
-      }))
+      if (user){
+        return res.status(CONFLICT).json({
 
-      return User.create(props)
+          ok: false,
+          message: "User already exists"
+        })
+      } else {
```

Comment:
Best practice is to return early on errors/failed validations, and then let the function's default behaviour simply run if conditions are met (rather than wrapping it all in an `else`). 

Read more: https://medium.com/swlh/return-early-pattern-3d18a41bba8